index.js

```diff
@@ -27,100 +22,80 @@ function readFile(input) {
   izborDiv.style.display = "none";
 
   reader.onload = function () {
-    tekst.innerText = reader.result;
-    let a = reader.result.split("\r\n");
+    text.innerText = reader.result;
+    let records = reader.result.split("\r\n");
 
-    for (let i = 0; i < a.length - 1; i++) {
-      Baza[parseInt(a[i].split(",").shift())] = a[i];
-
-      cenaLager =
-        cenaLager +
-        parseFloat(a[i].split(",")[4]) * parseFloat(a[i].split(",")[7]);
-      cenaStvarno =
-        cenaStvarno +
-        parseFloat(a[i].split(",")[4]) * parseFloat(a[i].split(",")[6]);
-    }
-    localStorage.setItem("baza", JSON.stringify(Baza));
-    localStorage.setItem("lager", JSON.stringify(cenaLager));
-    localStorage.setItem("lagerStvarno", JSON.stringify(cenaStvarno));
+    records.forEach(record => {
+      let obj = parseRecord(record);
+      storage[obj.id] = obj;      
+      actualPrice += obj.price * obj.realStock;
+      stockPrice += obj.price * obj.stock;
+    });
+    updateStorage();
   };
 
   reader.onerror = function () {
     console.log(reader.error);
   };
 }
 
+function updateStorage() {
+  localStorage.setItem("baza", JSON.stringify(storage));
+  localStorage.setItem("lager", stockPrice);
+  localStorage.setItem("lagerStvarno", actualPrice);
+}
+
+// parse csv string line to an actual object
+function parseRecord(csvRecord) {
+  let record = csvRecord.split(",");
+  return {
+    id: parseInt(record[0]),
+    name: record[1],
+    // not sure something: record[2],
+    // not sure something: record[3],
+    price: parseFloat(record[4]),
+    // not sure something: record[5]
+    realStock: parseFloat(record[6]),
+    stock: parseFloat(record[7]),
+    originalCSV: csvRecord,
+  }
+}
```

Comment:
Najbitnija promena, umesto da pamtis objekte u csv format pa da stalno moras da da radis: `.split(',')` i `parseFloat`, trebalo bi to samo jednom u pocetku uraditi i posle raditi samo sa objektima lako se pristupa podacima kao sto su cena, lager, itd.

Reply:
Ili 2D niz. dugo sam razmisljao kako da ovo uradim. problem mmi je bio sto nisu podaci sa incrementovanim id-jem, vec su (bar kod moje zene) ad-hok. Tu se lupaju sifre artikla kako ona ili knjigovodja stigne. a stalno sam mozgao sta je procesoru lakse da obradi: niz objekata ili niz stringova (pa i niz nizova). Jedini short-cut koji sam mogao da smislim je da uzmem sta god je knjigovodja definisao i pustim da preskoceni brojevi budu void u nizu. Hence - stalna provera da li je nesto null